layouts/partials/alert.html

```diff
@@ -0,0 +1,49 @@
+{{ $color := index . 0 }}
+{{ $message := index . 1 }}
+{{ $link := index . 2 }}
+
+<!-- creates a "mostly" unique id if there's multiple alerts -->
+{{ $uniqueness := replace $message " " "" }}
+
+<div class="alertBanner" id="{{ $uniqueness }}" style="background-color: {{ $color }}">
+  {{ if $link }}
+    <a class="alertBannerText" href="{{ $link }}">{{ $message }}</a>
+    <span class="closeAlert"><i class="fa fa-times" aria-hidden="true" onClick="closeAlert({{ $uniqueness }})"></i></span>
+  {{ else }}
+    <span class="alertBannerText">{{ $message }}</span>
+    <span class="closeAlert"><i class="fa fa-times" aria-hidden="true" onClick="closeAlert({{ $uniqueness }})"></i></span>
+  {{ end }}
+</div>
+
+<script>
+  alerts = Array.prototype.slice.call(document.getElementsByClassName("alertBanner"));
+  currentBanner = $(document.getElementById({{ $uniqueness }}));
+  // 10px is a small hack for different screen sizes
+  // semi-breaks if you're scaling the page large to small, but is fine on refresh
+  height = currentBanner.height() + 10;
+
+  // stack the alerts as needed
+  for (i = 1; i <= alerts.length; i++) {
+    lastLocation = parseInt($(alerts[i]).css('top'), 10);
+    newLocation = lastLocation + height;
+    currentBanner.css('top', newLocation + "px");
```

Comment:
Maybe we should a parent section named `alertBanners`, which contains the `width`, `top` & `position` properties and we just add `alertBanner` div elements to that section so we don't have to deal with this calculation.

Reply:
Additionally, maybe it would be easier to add the proper padding to that parent section so it does't go over the page navigation/title, like in this screenshot:

![](https://user-images.githubusercontent.com/1707663/38709907-75770316-3e72-11e8-9c16-661597a9cb89.png)

Reply:
You pointed me on such a good path here! It felt SO good to remove all that janky math.